aioftp/client.py

```diff
@@ -338,6 +341,188 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        def parse_rw(s):
+            mode = 0
+
+            if s[0] == 'r':
+
+                mode |= 4
+
+            elif s[0] != '-':
+
+                raise ValueError
+
+            if s[1] == 'w':
+
+                mode |= 2
+
+            elif s[1] != '-':
+
+                raise ValueError
+
+            return mode
+
+        mode = 0
+        mode |= parse_rw(s[0:2]) << 6
+        mode |= parse_rw(s[3:5]) << 3
+        mode |= parse_rw(s[6:8])
+
+        if s[2] == 's':
+
+            mode |= 0o4100
+
+        elif s[2] == 'x':
+
+            mode |= 0o0100
+
+        elif s[2] != '-':
+
+            raise ValueError
+
+        if s[5] == 's':
+
+            mode |= 0o2010
+
+        elif s[5] == 'x':
+
+            mode |= 0o0010
+
+        elif s[5] != '-':
+
+            raise ValueError
+
+        if s[8] == 't':
+
+            mode |= 0o1000
+
+        elif s[8] == 'x':
+
+            mode |= 0o0001
+
+        elif s[8] != '-':
+
+            raise ValueError
+
+        return mode
+
+    @staticmethod
+    def parse_ls_date(s):
```

Comment:
The point was to avoid parsing what we don't need to. We have the date in string format, so we just use it as is. But if you insist, I'm fine with using the library functions.